lib/logproto/logproto-text-server.c

```diff
@@ -291,6 +291,30 @@ log_proto_text_server_set_multi_line(LogProtoServer *s, MultiLineLogic *multi_li
   self->multi_line = multi_line;
 }
 
+static gboolean
+log_proto_text_server_restart_with_state(LogProtoServer *s, PersistState *persist_state, const gchar *persist_name)
+{
+  LogProtoTextServer *self = (LogProtoTextServer *) s;
+
+  gboolean res = log_proto_buffered_server_restart_with_state(s, persist_state, persist_name);
+  if (!res)
+    return FALSE;
+
+  LogProtoBufferedServerState *state = log_proto_buffered_server_get_state(&self->super);
+  const guchar *buffer_start = self->super.buffer + state->pending_buffer_pos;
+  gsize buffer_bytes = state->pending_buffer_end - state->pending_buffer_pos;
+
+  if (buffer_bytes > 0)
+    {
+      const guchar *eom = self->find_eom(buffer_start, buffer_bytes);
+      if (eom)
+        self->cached_eol_pos = eom - self->super.buffer;
```

Comment:
Don't we need to adjust `cached_eol_pos` based on what `log_proto_text_server_accumulate_line()` does?

Reply:
I think we don't but maybe it would be easier to review this IRL or in a call.

My understanding is that cached_eol_pos contains the next "unprocessed" NL character that needs to be accumulated into a multi-line buffer.

Also, cached_eol_pos != 0 means that we don't need to poll, we can immediately try to extract a line from the buffer (see the prepare() method).

If restart_with_state() find the next EOL in our buffer, then:
1) prepare() would return TRUE, indicating that we can go ahead and start extracting the line text
2) fetch_from_buffer() will use this as the first line and continue with accumulating the next line
3) we would loop as long as we have newline characters in the buffer (or if the line is complete).

With that I think this works correctly. But I'll do some more testing.